### PR TITLE
separate data collection from reporting

### PR DESCRIPTION
**Pull request by zsoldosp**
the combining the output from multiple  runs (e.g.: parallel execution) should be in a different MR, as it is already large enough

### CODE REVIEW

File: tests/testapp/tests/test_serialization.py [hunk 1]
```diff
@@ -46,6 +46,12 @@ def test_writer_writes_collected_results_fired_between_statt_stop(tmpfilepath):
     reader = serializer.Reader(tmpfilepath)
     deserialized = reader.read_all()
     assert deserialized == [(WithId('after start'), [2], {'after': 'start'})]
+    writer.end()  # dump daa again
```
Comment (kairichard) [inline tests/testapp/tests/test_serialization.py]: s/daa/data/

**Comment by kairichard** [comment]
Looks good to me.

**Comment by zsoldosp** [comment]
@kairichard this has been merged, yet https://pypi.python.org/pypi/django-performance-testing/ still only has the previous version (0.6.1). What happened w/ the release train?

**Comment by kairichard** [comment]
@zsoldosp There seems to be a problem with pusblishing the package -> 

`HTTPError: 410 Client Error: Gone (This API has been deprecated and removed from legacy PyPI in favor of using the APIs available in the new PyPI.org implementation of PyPI (located at https://pypi.org/). For more information about migrating your use of this API to PyPI.org, please see https://packaging.python.org/guides/migrating-to-pypi-org/#uploading. For more information about the sunsetting of this API, please see https://mail.python.org/pipermail/distutils-sig/2017-June/030766.html) for url: https://pypi.python.org/pypi/`

**Comment by kairichard** [comment]
@zsoldosp  Will look into that ASAP

**Comment by kairichard** [comment]
@zsoldosp Here is the reason the pipeline failed -> https://mail.python.org/pipermail/distutils-sig/2017-June/030766.html

**Comment by kairichard** [comment]
@zsoldosp the problem is found and fixed

**Pull request merged**